File: Grind169/Week2/383-Ransom-Note/solution.py
from collections import defaultdict

class Solution:
    def canConstruct(self, ransomNote: str, magazine: str) -> bool:
        mapping=defaultdict(int)

        for i in ransomNote:
            mapping[i]+=1

        for j in magazine:
            mapping[j]-=1

        for key,value in mapping.items():
            if value>0:
                return False
            
        return True

File: Grind169/Week2/383-Ransom-Note/solution.cpp
#include <string>
#include <unordered_map>

using namespace std;


class Solution {
public:
    bool canConstruct(string ransomNote, string magazine) {
        unordered_map<char,int>map;

        for (int i =0;i<ransomNote.size();i++){
            if (map.find(ransomNote[i])==map.end()){
                map.insert(pair<char,int>(ransomNote[i],0));
            }
            else {
                map[ransomNote[i]]+=1;
            }
        }

        for (int j=0;j<magazine.size();j++){
            if (map.find(magazine[j])!=map.end()){
                map[magazine[j]]-=1;
            }
        }

        for(const auto&n:map){
            if (n.second>0){
                return false;
            }
        }

        return true;
    }
};